### PR TITLE
WIP: add re-use for client connection on import  

### PR DESCRIPTION
**Pull request by tgruben**
added "caching" for direct client connections in order to more efficiently handle keep-alive connections

### CODE REVIEW

File: client.go [hunk 1]
```diff
@@ -61,6 +61,8 @@ type Client struct {
 	cluster *Cluster
 	host    *URI
 	client  *http.Client
+	slice_clients map[string]*Client
```
Comment (travisturner) [inline client.go]: can we call this something like `clientCache` (also, underscores are not typically used in go)

and with that change, in `getDirectClient()` I would change `cacheClient` to just `client` and change `newClient :=` to `client =`.

**Comment by jaffee** [comment]
I wrote up a separate issue for some (potentially) more invasive changes, but really there should only be one net/http.Client object per host in the cluster, and it should be re-used for all requests to that host throughout the life of the program: https://github.com/pilosa/go-pilosa/issues/48

**Comment by yuce** [comment]
The test for the cache is here: https://github.com/tgruben/go-pilosa/pull/1

**Comment by yuce** [comment]
I think we should take the client cache out of the client and make it its own struct, something like `ClientManager`. Otherwise the address of client object which contains the cache is not in the cache:
```go
client := DefaultClient()
client2, _ := client.getDirectClient("foo.bar:10101")
if client != client2 {
    // client itself is not cached
}
```

**Pull request merged**